### DISCUSSION TITLE
Meu site em JAM Stack... Mas como vou fazer sites para clientes em JAM Stack??

### DISCUSSION THREAD

**Discussion by wilsonneto-swe** [frontendbr/forum#1706] [Perguntas e Respostas]
Fala galera, beleza?

Estou renovando meu blog, vou mover ele de WP para JAM Stack com Next SSG + ReactJS + API em Node + BD em Json com lowdb...

Meu fluxo será eu inserir um arquivo markdown em uma pasta, cadastrar os metadados dele na base que estará em Json puro com LowDB (para deixar tudo no github msm...), após inserir dou um push e o CI da Netlify através do Next/SSG builda e já manda a atualização pro ar.

Mas minha dúvida é: Como eu faria um site para um cliente assim... vocês já fizeram, como fizeram? Digo porque o cliente vai querer inserir um post, ele provavelmente não vai saber fazer o markdown nem gerar o build... Como eu poderia fazer? 

Acredito que criar um painel para o cliente subir o site seria meio que a contra-mão da JAM Stack já que eu precisaria de um servidor de backend né? Por isso fiquei pensando... Para nós programadores, ok... mas para cliente que não mandam muito bem em código acho que complica, né?...

Gostaria de opiniões, abraços!

**Comment by victorferreira** [comment]
Você pode usar um CMS normalmente, inclusive o próprio Wordpress. Uma opção é usar algo como o [Contentful](https://www.contentful.com) CMS do próprio [Netlify](https://netlify.com), que também tem a opção [self-hosted](https://www.netlifycms.org).

**Comment by wilsonneto-swe** [comment]
Valeu, este meu vou fazer a api em mesmo e salvar como json / lowdb, pra já soltar logo rs 
Já para os de clientes vi algumas opções como Strapi e Cockpit (Netlify CMS tbm, mas os outros dois me chamaram mais a atenção...)

Pensando no Strapi e Cockpit, alguém sabe se eles tem alguma configuração para assim que o cliente /adm salvar o novo post ou categoria, ele já startar um build no netlify/github actions?

**Comment by victorferreira** [comment]
Posso estar errado, mas se você usar o Stapi (ou algum outro CMS) não precisa refazer o build a cada novo post. O conteúdo passa a ser oferecido pela API e não é mais estático.

**Comment by wilsonneto-swe** [comment]
Isso, na vdd eu teria a opcao, e preciso que seja totalmente estatico, por isso vou usar o next como SSG e nao SSR... Estou cogitando ate usar o netlifyCms, assim nao preciso de servidor para nada, so a hospedagem dos sites estaticos mesmo... Esta e minha intencao...

**Comment by SrVieira** [comment]
Como já informado eu iria de NetlifyCMS, usar Strapi ou outro CMS-JS tu teria que ter duas aplicações rodando caso alimente sua aplicação final (front) com os dados vindo destes via API. Pois se tu está servindo sua aplicação através do Netlify não teria com dados dinamicos diretamerte.

**Comment by wilsonneto-swe** [comment]
@SrVieira vou me informar melhor sobre ele, mas em primeiro momento minha dúvida é, como ele guardará os artigos do meu blog, saberia me falar? Já usa e está gostando? E ele daria certo com clientes, ou é bem focado em devs?

(Já estou com a documentação aqui, na verdade vou fazer meu blog com minha arquitetura com os dados em um json mesmo, mas vou refazer ele com strapi, com cockpit e com netlifyCms, para entender todos os prós e contras de cada para assim quando precisar para algum cliente saber a ferramenta certa a usar... Em meu cenário/meu blog, não quero servidor msm, se eu usar o strapi vai ser apenas em minha máquina local para usar de api para gerar o build SSG do next... Subir vou subir apenas arquivos estáticos, mas no caso de sites para clientes futuramente, ai eu poderia subir outra app como strapi se ela me ajudar a entregar mais rápido)
